piker/ui/_curve.py

```diff
@@ -83,21 +180,47 @@ def update_from_array(
         #     # self.path.moveTo(new_x[0], new_y[0])
         #     self.path.connectPath(old_path)
 
-        if append_length:
-            # print(f"append_length: {append_length}")
-            new_x = x[-append_length - 2:-1]
-            new_y = y[-append_length - 2:-1]
-            # print((new_x, new_y))
+        elif append_length > 0:
+            if self._step_mode:
+                new_x, new_y = step_path_arrays_from_1d(
+                    x[-append_length - 2:-1],
+                    y[-append_length - 2:-1],
+                )
+                new_x = new_x[1:]
+                new_y = new_y[1:]
+
+            else:
+                # print(f"append_length: {append_length}")
+                new_x = x[-append_length - 2:-1]
+                new_y = y[-append_length - 2:-1]
+                # print((new_x, new_y))
 
             append_path = pg.functions.arrayToQPath(
                 new_x,
                 new_y,
-                connect='all'
+                connect='all',
+                # finiteCheck=False,
             )
-            # print(f"append_path br: {append_path.boundingRect()}")
-            # self.path.moveTo(new_x[0], new_y[0])
-            # self.path.connectPath(append_path)
-            self.path.connectPath(append_path)
+
+            path = self.path
+
+            # other merging ideas:
+            # https://stackoverflow.com/questions/8936225/how-to-merge-qpainterpaths
+            if self._step_mode:
+                if self._fill:
```

Comment:
We need to try out work from: https://github.com/pyqtgraph/pyqtgraph/pull/2032

Reply:
Ok, tried it and it's still horrid. Gonna comment out this code and leave comments but honestly i think the consensus from the ranks is that that unfilled (except current bar) looks better anyway.

`pg` is using the same pain call:
https://github.com/pyqtgraph/pyqtgraph/pull/2032/files#diff-0110e152aa13c81b6b8b9c227f40b855c44cc86cf1fb5c9861d57e630898d48fR764 which is the super slow part afaict.

We can always take a look at this down the road if there's a big want from future users.